styles/color_select.h

```diff
@@ -0,0 +1,56 @@
+#ifndef STYLES_COLOR_SELECT_H
+#define STYLES_COLOR_SELECT_H
+
+// Usage: ColorSelect<SELECTION, TRANSITION, COLOR1, COLOR2, ...>
+// SELECTION: function
+// TRANSITION: transition
+// COLOR1, COLOR2, ...:  COLOR
+// Return value: COLOR
+// Decides what color to return based on the current selection.
+// The returned color will be selection % N (where N is the number of colors arguments).
+// When the selection changes, the transition will be used to change from the old color to the new color.
+
+template<class SELECTION, class TRANSITION, class... COLORS>
+class ColorSelect {
+public:
+  ColorSelect() {
+    if (is_same_type<FUNC, Variation>::value) {
+      BladeBase::HandleFeature(HANDLED_FEATURE_CHANGE_TICKED);
+    }
+  }
+  void run(BladeBase* blade) {
+    f_.run(blade);
+    colors_.run(blade);
+    uint8_t selection = f_.getInteger(0);
```

Comment:
this needs to be something like this:
int f = f_.getInteger(0);
while (f < 0) f += sizeof...(COLORS) << 8;
uint8_t selection = f % sizeof...(COLORS);

Reply:
thanks, I have updated